src/main.c

```diff
@@ -202,7 +237,10 @@ int collect_pids(pid_t **rpids) {
 
 		for (char **p = config.parent_commands.list; *p; p++) {
 			if(!strcmp(comm, *p)) {
-				pids[n_pids++] = pid;
+				if (!force && _already_in_systemd(pid, config.slice))
```

Comment:
Given possible failure in `_already_in_systemd`, I'd propagate it through whole `collect_pids` (similar to `read_stat`) failure to be on the safe side when such an unexpected error happens.

Reply:
Changed in the new commit.